src/uk/honeymanga/src/eu/kanade/tachiyomi/extension/uk/honeymanga/HoneyManga.kt

```diff
@@ -0,0 +1,229 @@
+        return GET(mangaUrl, headersBuilder().build())
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val chapterId = chapter.url.substringAfterLast('/')
+        val url = "$API_URL/chapter/frames/$chapterId"
+        return GET(url, headersBuilder().build())
+    }
+
+    // ----- parse -----
+
+    override fun popularMangaParse(response: Response): MangasPage = parseAsMangaResponseDto(response)
+
+    override fun latestUpdatesParse(response: Response): MangasPage = parseAsMangaResponseDto(response)
+
+    override fun searchMangaParse(response: Response): MangasPage = parseAsMangaResponseArray(response)
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        return makeSManga(response.asClass())
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val result = response.asClass<List<HoneyMangaChapterDto>>()
+        return result.map {
+            SChapter.create().apply {
+                url = "$API_URL/chapter/frames/${it.id}"
+                name = "Vol. ${it.volume} Ch. ${it.chapterNum}" + (if (it.subChapterNum == 0) "" else ".${it.subChapterNum}")
+                date_upload = parseDate(it.lastUpdated)
+            }
+        }
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        return response.asClass<HoneyMangaChapterPagesDto>().resourceIds.toList().map {
+            Page(index = it.first.toInt(), url = "", imageUrl = "$IMAGE_STORAGE_URL/${it.second.jsonPrimitive.content}")
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = ""
+
+    override fun imageRequest(page: Page): Request {
+        val newHeaders = headersBuilder().build()
+        return GET(page.imageUrl!!, newHeaders)
+    }
+
+    override fun getMangaUrl(manga: SManga): String = manga.url
+
+    override fun getChapterUrl(chapter: SChapter): String = chapter.url
+
+    override fun fetchImageUrl(page: Page): Observable<String> = Observable.just(page.imageUrl!!)
+
+    companion object {
+
+        // constants and utils
+
+        private const val API_URL = "https://data.api.honey-manga.com.ua"
+
+        private const val SEARCH_API_URL = "https://search.api.honey-manga.com.ua"
+
+        private const val IMAGE_STORAGE_URL = "https://manga-storage.fra1.digitaloceanspaces.com/public-resources"
+
+        private val userAgent = System.getProperty("http.agent")!!
+
+        private const val ISO_DATE = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
+
+        private val dateFormat = SimpleDateFormat(ISO_DATE, Locale.ROOT)
+
+        private fun parseDate(dateAsString: String): Long = dateFormat.parse(dateAsString)?.time ?: 0L
+
+        private const val DEFAULT_PAGE_SIZE = 30
+
+        private val json = Json {
+            isLenient = true
+            ignoreUnknownKeys = true
+        }
+
+        private inline fun <reified R : Any> Response.asClass() = run { json.decodeFromString<R>(body.string()) }
```

Comment:
Use `use` instead so it closes the `Response` as needed after using it.

```suggestion
        private inline fun <reified R : Any> Response.asClass() = use { json.decodeFromString<R>(body.string()) }
```